book/src/01-02-instances-etc.md

```diff
@@ -0,0 +1,190 @@
+# Instances, Adapters, and Devices
+
+## Initialization
+
+The first example will simply initialize the library and print some information about the available graphics devices on the system.
+Initialization is performed by constructing an `Instance` and specifying the desired backend(s).
+Create a new subdirectory `src/bin/hello/`, and in `src/bin/hello/main.rs`, add the following:
+
+```rust,no_run,noplayground
+fn main() {
+    let instance = wgpu::Instance::new(wgpu::BackendBit::PRIMARY);
+}
+```
+
+This initializes `wgpu` with any of the available stable backends.
+If you want to target a specific backend, you can do that too&mdash;see the docs for [`BackendBit`].
+
+> #### What's `BackendBit::PRIMARY`?
+>
+> `BackendBit` is a bitflag type that specifies the desired backend(s).
+> `BackentBit::PRIMARY` is a combination of all backends for which `wgpu` provides "first-class" support&mdash;namely Vulkan, D3D12, and Metal.
+> Experimental and incomplete backends (D3D11 and OpenGL) are designated `BackendBit::SECONDARY`.
+
+## Selecting an adapter
+
+Now that you have an `Instance`, you can query for the available adapters on a system with `Instance::enumerate_adapters`.
+This returns an iterator over a series of `Adapter` objects.
+Try printing out the info for each adapter (`AdapterInfo`), like this:
+
+```rust,no_run,noplayground
+fn main() {
+    // let instance = ...
+
+    for adapter in instance.enumerate_adapters(wgpu::BackendBit::PRIMARY) {
```

Comment:
I think we should focus on the API that is portable to the Web. `enumerate_adapters` can be polyfilled by returning a single adapter, but essentially `request_adapter` is most portable